torch_xla/experimental/tpu.py

```diff
@@ -70,17 +71,24 @@ def _get_metadata(key: str) -> str:
   return resp.text
 
 
-def process_bounds_size(default: int = 1) -> int:
-  """Returns number of processes across all TPU hosts."""
+def process_bounds_size() -> Optional[int]:
+  """Returns number of processes across all TPU hosts, or None if unknown."""
   process_bounds = xu.getenv_as(xenv.TPU_PROCESS_BOUNDS, str)
-  return MeshShape.from_string(
-      process_bounds).size if process_bounds else default
+  return MeshShape.from_string(process_bounds).size if process_bounds else None
 
 
-def num_local_processes(local_chips: int = 4) -> int:
+def num_available_chips() -> int:
+  """Returns the number of local chips in /dev/"""
```

Comment:
qq: does it apply to v3/v2? Asking it because iirc there are 4 chips on a v3-8 donut but we can see /dev/accel0 up to /dev/accel7.

Reply:
Actually, each `/dev/accel*` represents one chip, not one core, so only 4 devices will show up in `/dev/`. For example, on a TPU v2:

```
$ ls /dev/accel*
/dev/accel0  /dev/accel1  /dev/accel2  /dev/accel3
```